### PR TITLE
nitch: fix nerdfont icons

### PR DESCRIPTION
**Pull request by michaelBelsanti**
###### Description of changes

Applied a patch to fix the nerd font icons. Updated the author name.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — CHANGES_REQUESTED** [review]
A couple of minor things to change before merging...

File: pkgs/tools/misc/nitch/default.nix [hunk 1]
```diff
@@ -4,8 +4,8 @@ nimPackages.buildNimPackage rec {
   version = "0.1.6";
   nimBinOnly = true;
   src = fetchFromGitHub {
-    owner = "unxsh";
-    repo = "nitch";
+    owner = "ssleert";
+    repo = pname;
```
[the remark this answers is not in the document]
Comment (michaelBelsanti) [reply pkgs/tools/misc/nitch/default.nix:8]: Will change. Is there a reason this is preferred?

File: pkgs/tools/misc/nitch/default.nix [hunk 2]
```diff
@@ -15,6 +15,10 @@ nimPackages.buildNimPackage rec {
       url = "https://github.com/unxsh/nitch/commit/6831cf96144f58c4da298a0bc9b50d33056f6c08.patch";
       sha256 = "sha256-uZUzUBLHBsssNqDxZ0NuTRMN9/gBxIlIiGgQkqCqEFc=";
     })
+    (fetchpatch {
+      url = "https://github.com/ssleert/nitch/commit/ff10dea47dbc4d9736bd839b87869f53166cf395.patch";
+      sha256 = "sha256-n7JtI1Td0xMqhuScnMvxTkbY2I+l+FcUTz7aytxFasE=";
```
[the remark this answers is not in the document]
Comment (michaelBelsanti) [reply pkgs/tools/misc/nitch/default.nix:20]: Will change. Is hash generally preferred over sha256 in nixpkgs?

File: pkgs/tools/misc/nitch/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/tools/misc/nitch/default.nix:8]: I do not know the very exact reason, but I saw the same comment from @SuperSandro2000 , I'm pretty sure he will tell you why.

File: pkgs/tools/misc/nitch/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/tools/misc/nitch/default.nix:20]: Yes.

**Pull request merged**